### PR TITLE
Add Ray serve example using AWS neuron-cores

### PR DESCRIPTION
**Pull request by chappidim**
*Description of changes:*

This example helps customers to launch a cluster for serving inference model using Ray serve. Apart from Ray documentation, this example provides
 1. Ray cluster with trainium instance launch and appropriate neuron-sdk commands
 2. Ray Serve example using Open LLAMA to compile and deploy the model.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
> I confirm

### CODE REVIEW

File: examples/serve-inference/aws_neuron_core_inference_serve.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+from fastapi import FastAPI  # noqa
+from ray import serve  # noqa
+
+import torch # noqa
+import os
+from transformers import AutoTokenizer # noqa
+
+app = FastAPI()
+
+hf_model = "openlm-research/open_llama_3b"
```
Comment (jianyinglangaws) [inline examples/serve-inference/aws_neuron_core_inference_serve.py:10]: Can we change this to llama2 instead of open-llama?

**Review by jianyinglangaws — COMMENTED** [review]
It could be better to use jupyter notebook format so it can be more interactive to users.

File: examples/serve-inference/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## Setup 
+* Clone this repo to run the example on your local laptop:
+```
+git clone https://github.com/aws-samples/aws-samples-for-ray
+
+cd aws-samples-for-ray/examples/serve-inference
+```
+
+* Replace subnet and security-group where you intend to launch the cluster in `cluster-inference-serve.yaml`
```
Comment (jianyinglangaws) [inline examples/serve-inference/README.md:14]: Maybe explain how to create the subnet and security group here. You can just add the link to AWS docs.

File: examples/serve-inference/aws_neuron_core_inference_serve.py (hunk 1 above)
Comment (chappidim) [reply examples/serve-inference/aws_neuron_core_inference_serve.py:10]: I would prefer to have llama2 (commented). Default to open_llama as we don't want customers to wait for license approvals.

**Pull request merged**